scripts/load_mentors.js

```diff
@@ -113,16 +115,26 @@ exports.handler = async (event) => {
     const { id: mentorId } = result;
     if (mentorMap.get(mentorId) === undefined) {
       deletedMentorIds.push(mentorId);
-    } else if (
-      mentorMap.get(mentorId).last_modified_time !== result.last_modified_time
-    ) {
-      modifiedMentorIds.push(mentorId);
+    } else {
+      leftoverMentorIdsSet.delete(mentorId);
+      if (
+        mentorMap.get(mentorId).last_modified_time !== result.last_modified_time
+      ) {
+        modifiedMentorIds.push(mentorId);
+      }
+    }
+  }
+  for (const mentorId of leftoverMentorIdsSet) {
+    const mentor = mentorMap.get(mentorId);
+    if (mentor.name) {
+      addedMentorIds.push(mentorId);
     }
   }
```

Comment:
Rather than initialising `leftoverMentorIdsSet` to all existing mentor IDs from Airtable, deleting away the unmodified mentor IDs from Elastic Cloud, and then creating a new variable `addedMentorIds` with all the remaining mentor IDs, perhaps we can simplify the logic to the following:
1. Compile `Set` of deleted mentor IDs (`deletedMentorIds`) and unmodified mentor IDs (`unmodifiedMentorIds`).
2. Run through entire list of Airtable mentors and filter based on whether they're deleted or unmodified, and if they have a `mentor.name`:

```
const indexMentors = Array.from(mentorMap.entries())
    .filter([mentorId, mentor] => !deletedMentorIds.has(mentorId) && !unmodifiedMentorIds.has(mentorId) && mentor.name);
```
3. Amend `for` loop below so that it uses the mentors above and won't need to call `mentorMap.get(id)`.

Reply:
This way I think the logic is easier to follow, as compared to dealing with multiple `for` loops.

Reply:
Alright I should have just made the necessary edits. I removed the `deletedMentorIds` part as `mentorMap` (from the Airtable) is not supposed to have any deleted entries in the first place, so 1 less place to check